### DISCUSSION TITLE
Is there something like `log_marginal_likelihood()` ?

### DISCUSSION THREAD

**Discussion by MJimitater** [cornellius-gp/gpytorch#1679] [Q&A] [answered]
Hi folks, thank you for pushing forward the developments with v.1.5.0. 🎉

I was wondering: in sklearn after fitting a GP classifier, one can call the log marginal likelihood simply by `gp.log_marginal_likelihood()`.

Is there something like that in GPyTorch?

Thanks

**Comment by gpleiss** [comment, marked as answer]
Are you looking for the log marginal likelihood of a GP **classifier** or a regressor? For a regressor, call:

```python
model.likelihood(model(train_x)).log_prob(train_y)
```

**Reply by MJimitater** [reply]
Thanks! Indeed Im looking for the corresponding one of a **GP classifier**, is there something like it?

**Reply by gpleiss** [reply]
It's not possible to compute it exactly, since inference is intractable. With a variational model, you can get the variational lower bound via:

```python
mll = gpytorch.mlls.VariationalELBO(likelihood, model)
lower_bound = mll(model(train_x), train_y)
```
